### ISSUE TITLE
Very low FPS (< 10) in Song selection

### ISSUE THREAD

**Issue by cRaZy-bisCuiT** [UltraStar-Deluxe/USDX#320]
### Actual behaviour
When trying to select a song, a lot of them cause the FPS to drop below 10. It's very odd, since when I start the song the FPS go back to 60.

### Expected behaviour
Selecting a song shouldn't make FPS drop that much.

### Steps to reproduce

1. Open the Game
2. Select Players
3. Skimming through the song selection

### Details

Provide some additional information:

- USDX version: latest git as of 2017_11_09
- Operating System + version: 4.13.11-1-ARCH
- Graphics:

▶ lspci -nnk | grep -i VGA -A2 
01:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/580] [1002:67df] (rev c7)
	Subsystem: Micro-Star International Co., Ltd. [MSI] Radeon RX 480 [1462:3413]
	Kernel driver in use: amdgpu
~                                                                                                                                                                                            
▶ glxinfo | grep 'OpenGL version string' 
OpenGL version string: 3.0 Mesa 17.2.4

The same happens on Raspberry Pi as well as Core2Duo + Intel GMA 4500MHD in Ubuntu 17.04.4

**Comment by cRaZy-bisCuiT** [comment]
The issue still persists. It does not make a difference if I use a system with Intel 3570K@4,4Ghz and 16 GB RAM or a RPI4: Bad performance is hit in the song selection whenever the library seems to big.

One new observation: Only 700 Mb / 4 GB RAM are used when running the game. Is the RAM usage limited somewhere? I feel like the game could not really cache the whole database due to RAM usage limitations.

**Comment by s09bQ5** [comment]
According to OProfile it's TCatSongs.VisibleIndex that kills the performance:
```
Counted CPU_CLK_UNHALTED events (Clock cycles when not halted) with a unit mask of 0x00 (core_p Core cycles when core is not halted) count 100000
samples  %        image name               app name                 symbol name
1428511  44.5100  vmlinux                  vmlinux                  intel_idle
800926   24.9555  ultrastardx              ultrastardx              USONGS$_$TCATSONGS_$__$$_VISIBLEINDEX$LONGINT$$LONGINT
376580   11.7336  libavcodec.so.56.60.100  ultrastardx              /usr/lib64/libavcodec.so.56.60.100
68593     2.1372  libavutil.so.54.31.100   ultrastardx              /usr/lib64/libavutil.so.54.31.100
42660     1.3292  libnvidia-glcore.so.340.93 ultrastardx              /usr/lib64/libnvidia-glcore.so.340.93
40462     1.2607  libavformat.so.56.40.101 ultrastardx              /usr/lib64/libavformat.so.56.40.101
32895     1.0250  libc-2.18.so             ultrastardx              __GI___strcmp_ssse3
18619     0.5801  libm-2.18.so             ultrastardx              __sin_sse2
13551     0.4222  libc-2.18.so             ultrastardx              _int_malloc
12652     0.3942  libc-2.18.so             ultrastardx              __memcpy_ssse3
10608     0.3305  ultrastardx              ultrastardx              USCREENSONG$_$TSCREENSONG_$__$$_SETROULETTESCROLL
10552     0.3288  ultrastardx              ultrastardx              USCREENSONG$_$TSCREENSONG_$__$$_SETROULETTESCROLLREFRESH
```

It should be easy to optimize this by saving the position of a song in the list of visible songs inside TSong.

**Comment by s09bQ5** [comment]
Please test if this is fixed by #534
